src/core/or/connection_or.c

```diff
@@ -1364,10 +1366,34 @@ void
 connection_or_connect_failed(or_connection_t *conn,
                              int reason, const char *msg)
 {
+  const or_options_t *options = get_options();
+
   control_event_or_conn_status(conn, OR_CONN_EVENT_FAILED, reason);
-  if (!authdir_mode_tests_reachability(get_options()))
+  if (!authdir_mode_tests_reachability(options))
     control_event_bootstrap_prob_or(msg, reason, conn);
   note_or_connect_failed(conn);
+
+  if (options->ClientAutoIPv6ORPort) {
```

Comment:
This option check is good for an experimental feature. When the feature stops being experimental, we might always want to keep this state, even if the option is off.